src/main/java/uk/gov/hmcts/sptribs/common/event/page/RepresentativeDetails.java

```diff
@@ -29,4 +35,25 @@ public void addTo(PageBuilder pageBuilder) {
             .mandatory(CicCase::getRepresentativeAddress, "cicCaseRepresentativeContactDetailsPreference = \"Post\"")
             .done();
     }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore) {
+        final CaseData data = details.getData();
+        final List<String> errors = new ArrayList<>();
+
+        if (null != data.getCicCase() && null != data.getCicCase().getRepresentativeAddress()) {
+            if (StringUtils.isEmpty(data.getCicCase().getRepresentativeAddress().getCountry())) {
```

Comment:
If RepresentativeAddress is null, does that mean there should be an error added? Just a thought.

Reply:
No, it can address or email. We need to check for Country/post code only if Address is selected.